osbenchmark/worker_coordinator/worker_coordinator.py

```diff
@@ -1675,6 +1678,7 @@ async def execute_single(runner, opensearch, params, on_error):
             total_ops_unit = "ops"
             request_meta_data = {"success": True}
     except opensearchpy.TransportError as e:
+        request_context_holder.on_client_request_end()
         # we *specifically* want to distinguish connection refused (a node died?) from connection timeouts
         # pylint: disable=unidiomatic-typecheck
         if type(e) is opensearchpy.ConnectionError:
```

Comment:
Generally, it is better to catch any exceptions raised by an API call at the point where the call is made, rather than up the call stack.  That way, the caller can tailor the exceptions handled precisely, for instance, a _search_ call may raise a different set of exceptions as compared to a _force-merge_.  Otherwise, someone looking at the higher level call needs to dig down possible flows of control to identify exceptions that might get raised.

The current implementation of `execute_single()` is legacy in that it assumes it is tied to the ES/OS API.  In the future, we will support other clients, that might raise other exceptions.  Therefore, the strategy noted above would be preferable to the one implemented above.

BTW, the `execute_singe()` routine should indeed capture exceptions, but they should be more general, just to handle the eventuality that lower-level modules have a bug and missed catching something they should.

In summary, the change above is functionally correct, but it is recommended that exceptions be caught at the point of call.

Reply:
I understand your point, @gkamat. I'm not introducing new exception handling; I'm simply addressing the issue mentioned in https://github.com/opensearch-project/opensearch-benchmark/pull/450#issuecomment-1936836969. 

And I believe the error handling as said [here](https://github.com/opensearch-project/opensearch-benchmark/pull/462#discussion_r1488655383) is unrelated to my [PR](https://github.com/opensearch-project/opensearch-benchmark/pull/450)

Reply:
The proposed change requires exception handling for the client time computation functionality, else there is a KeyError that gets raised.  This should be caught at the point of the call, rather than up the stack.

The overall change does not need to be reverted; just exception handling for the current functionality needs to be added at each API call, rather than relying on the existing checks.  Ideally, the checks in `execute-single` might be moved down, but that is not a prerequisite.